core/lib/spree/money.rb

```diff
@@ -23,9 +32,33 @@ def from_money(money)
 
     delegate :cents, :currency, :to_d, to: :money
 
-    # @param amount [#to_s] the value of the money object
-    # @param options [Hash] the options for creating the money object
-    # @option options [String] currency the currency for the money object
+    # @param amount [Money, #to_s] the value of the money object
+    # @param options [Hash] the default options for formatting the money object See #format
+    def initialize(amount, options = {})
+      if amount.is_a?(::Money)
+        @money = amount
+      else
+        currency = (options[:currency] || Spree::Config[:currency])
+        if amount.to_s =~ /\A-?\d+(\.\d+)?\z/
```

Comment:
Please put this into a constant. We don't want "magic values" in our code base, right?